template_content/smart_contracts/helpers/util.py

```diff
@@ -1,8 +1,22 @@
 from pathlib import Path
 
+import tomllib
+
 
 def find_app_spec_file(output_dir: Path) -> str | None:
     for file in output_dir.iterdir():
         if file.is_file() and file.suffixes == [".arc32", ".json"]:
             return file.name
     return None
+
+
+def find_pyproject_version(pyproject_toml: Path) -> str | None:
```

Comment:
How about something like this instead

```
def find_pyproject_version(pyproject_toml: Path) -> str | None:
    """Finds the project version from a 'pyproject.toml' file.

    Args:
        pyproject_toml: The path to the 'pyproject.toml' file.

    Returns:
        The project version if found, otherwise None.
    """

    try:
        data = tomllib.load(pyproject_toml.open('rb'))
        return data.get('tool', {}).get('poetry', {}).get('version')
    except (FileNotFoundError, tomllib.TOMLDecodeError):
        return None 
  ```